### PR TITLE
532 Improve custom plot functionality

### PR DESCRIPTION
**Pull request by mknaranja**
## Merge Request - GuideLine Checklist 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
Always to be checked:
* [x] There is at least one issue associated with the pull request.
* [x] The branch follows the naming conventions as defined in the [git workflow](git-workflow).
* [x] New code adheres with the [coding guidelines](coding-guidelines)
* [x] Make sure that the [pre-commit linting/style checks pass](https://github.com/DLR-SC/memilio/wiki).

If functions were changed or functionality was added:
* [x] Tests for new functionality has been added
* [x] A local test was succesful

If new functionality was added:
* [x] There is appropriate **documentation** of your work. (use doxygen style comments)

If new third party software is used:
* [ ] Did you pay attention to its license? Please remember to add it to the wiki after successful merging.

If new mathematical methods or epidemiological terms are used:
* [ ] Are new methods referenced? Did you provide further documentation? Has the glossary been updated? 

The following questions are addressed in the documentation if need be: 
* [ ] Developers (what did you do?, how can it be maintained?)
* [ ] For users (how to use your work?)
* [ ] For admins (how to install and configure your work?)

* For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [x] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [x] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [x] There are appropriate **unit tests** and they pass.
* [x] The git history is clean and linearized for the merge request. All reviewers should squash commits and write a simple and meaningful commit message.
* [x] Coverage report for new code is acceptable.

### CODE REVIEW

**Comment by mknaranja** [comment]
@jubicker I slightly extended the customPlot functionality. However, when I wanted to write a simple unit test for it I came accross the problem that this does not work that simple. Could you have a look at that?

**Comment by mknaranja** [comment]
Thanks for adding the test @jubicker . If you agree, please merge it as reviewer.

File: pycode/memilio-epidata/memilio/epidata/customPlot.py [hunk 1]
```diff
@@ -37,36 +45,47 @@ def plotList(x, y, legend, title='', xlabel='', ylabel='', fig_name=''):
...
+
     if isinstance(x, pd.Series):
         x = x.values
-    ax.set_xticks(x[tick_range])
-    ax.set_xticklabels(x[tick_range], rotation=90, fontsize=8)
+    if xticks_idx == 'default':
+        tick_range = (np.arange(int(len(x) / 30) + 1)
+                      * 30)  # only plot every 30th day
+        tick_range[-1] -= 1  # adapt for last element
+        ax.set_xticks(x[tick_range])
```
Comment (jubicker) [inline pycode/memilio-epidata/memilio/epidata/customPlot.py:71]: Isn't tick_range a list and no indice? Should it be ax.set_xticks([x[tick_range[i]] for i in tick_range])?

File: pycode/memilio-epidata/memilio/epidata/customPlot.py [hunk 2]
```diff
@@ -37,36 +45,47 @@ def plotList(x, y, legend, title='', xlabel='', ylabel='', fig_name=''):
...
+        ax.set_xticklabels(x[tick_range], rotation=90, fontsize=8)
```
Comment (jubicker) [inline pycode/memilio-epidata/memilio/epidata/customPlot.py:72]: same as above

File: pycode/memilio-epidata/memilio/epidata/customPlot.py [hunk 3]
```diff
@@ -37,36 +45,47 @@ def plotList(x, y, legend, title='', xlabel='', ylabel='', fig_name=''):
...
+        ax.set_xticks([x[i] for i in xticks_idx])
+        ax.set_xticklabels([x[i] for i in xticks_idx],
+                           rotation=90, fontsize=10)
+
     if xlabel != '':
         ax.set_xlabel(xlabel, fontsize=12)
     if ylabel != '':
         ax.set_ylabel(ylabel, fontsize=12)
 
     if len(y) > 3:
```
Comment (jubicker) [inline pycode/memilio-epidata/memilio/epidata/customPlot.py:83]: why is a legend only added if len(y)>3?

**Review by jubicker — APPROVED** [review]

**Pull request merged**